spec/pundit_spec.rb

```diff
@@ -33,6 +33,26 @@
       expect(Pundit.authorize(user, [:project, comment], :update?)).to be(comment)
     end
 
+    it "returns the record when passed record with nested namespace " do
+      expect(Pundit.authorize(user, [:project, :admin, comment], :update?)).to be(comment)
+    end
+
+    it "can use headless policy" do
+      expect(Pundit.authorize(user, :publication, :create?)).to be_truthy
+    end
+
+    it "returns the policy name symbol when passed record with headless policy" do
+      expect(Pundit.authorize(user, :publication, :create?)).to be(:publication)
+    end
+
+    it "can use without a particular instance" do
+      expect(Pundit.authorize(user, Post, :show?)).to be_truthy
```

Comment:
This feels like quite a weak assertion - is truthiness a property we care about? 

Would it be better to make this an assertion about the return value (`"returns the... "`) and to expect the specific value we get in this case? (is it `Post`?)

Reply:
I see.

Reply:
But we have not 'can use without a particular instance' spec and 'can use headless policy'.
May I add it at this time?

Reply:
The thing for me is that I'm not sure what "can use" (or "can be used") is intending to test.

For me there are usually two types of test, which correspond to the two kinds of property which we care about our code having:
1. What's the return value (if we care)
2. What side effects happen (if any)

If we have one of these tests for a particular scenario/context then we're also implicitly testing that we "can use" the method in that context, since we're using the code. If we don't care about either 1 or 2, then probably the code doesn't need to exist 😉 

Am I making sense, or am I missing the point?

Reply:
Thanks! It became clear.
Certainly, 1 is testing that we "can use" the method.

Reply:
Cool, in that case I suggest it should be the same shape as the other specs which test a return value:
title: `it "returns..." do`
expectation: `expect(Pundit.authorize(...).to eq(<specific value>)`

Reply:
Yes
I left only test a return value.